arroyo-connectors/src/filesystem/source.rs

```diff
@@ -160,15 +176,50 @@ impl FileSystemSourceFunc {
         Ok(SourceFinishType::Final)
     }
 
+    async fn get_newline_separated_stream(
+        &mut self,
+        storage_provider: &StorageProvider,
+        path: String,
+    ) -> Result<Box<dyn Stream<Item = Result<String, UserError>> + Unpin + Send>, UserError> {
+        match &self.format {
+            Format::Json(_) => {
+                let stream_reader = storage_provider.get_as_stream(path).await.unwrap();
+
+                let compression_reader: Box<dyn AsyncRead + Unpin + Send> =
+                    match self.get_compression_format() {
+                        CompressionFormat::Zstd => {
+                            Box::new(ZstdDecoder::new(BufReader::new(stream_reader)))
+                        }
+                        CompressionFormat::Gzip => {
+                            Box::new(GzipDecoder::new(BufReader::new(stream_reader)))
+                        }
+                        CompressionFormat::None => Box::new(BufReader::new(stream_reader)),
+                    };
+                // use line iterators
+                let lines = LinesStream::new(BufReader::new(compression_reader).lines());
+                Ok(Box::new(lines.map(|string_result| {
+                    string_result.map_err(|err| {
+                        UserError::new("could not read line from stream", err.to_string())
+                    })
+                })))
+            }
+            other => Err(UserError::new(
+                "bad format",
+                format!("newline separated stream not supported for {:?}", other),
+            )),
+        }
+    }
+
     async fn get_record_batch_stream(
         &mut self,
         storage_provider: &StorageProvider,
         path: String,
+        out_schema: SchemaRef,
     ) -> Result<Box<dyn Stream<Item = Result<RecordBatch, UserError>> + Unpin + Send>, UserError>
     {
         match &self.format {
-            Format::Json(_) => Err(UserError::new("unsupported format:", "json")),
-            Format::Avro(_) => Err(UserError::new("unsupported format:", "avro")),
+            Format::Json(_) => Err(UserError::new("unsupported format", "json")),
```

Comment:
It seems like if you end up here this is due to a coding error, so maybe these should be panics?